### PR TITLE
[Lib] Add Memory and CPU usage monitoring support

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by pranavprakash20** [comment]
Output from one node

[root@ceph-pprakash-jue5j3-node1-installer ~]# cat ceph-mon_ceph-pprakash-jue5j3-node1-installer_configure_client_0.csv
ceph-pprakash-jue5j3-node1-installer_configure_client_0,,,
Time stamp,Process Name,Process ID,CPU Usage,Memory Usage
[2024-14-02 13:58:01],ceph-mon,76566,1.2,76.2812
[2024-14-02 13:58:06],ceph-mon,76566,1.2,76.3398
[2024-14-02 13:58:11],ceph-mon,76566,1.2,76.3398
[2024-14-02 13:58:17],ceph-mon,76566,1.2,77.1758
[2024-14-02 13:58:22],ceph-mon,76566,1.2,76.8711
[2024-14-02 13:58:27],ceph-mon,76566,1.2,76.8711
[root@ceph-pprakash-jue5j3-node1-installer ~]# ls
ceph-mgr_ceph-pprakash-jue5j3-node1-installer_Deploy_cluster_using_cephadm_0.csv
ceph-mgr_ceph-pprakash-jue5j3-node1-installer_Executes_RGW,_RBD_and_FS_operations_0.csv
ceph-mgr_ceph-pprakash-jue5j3-node1-installer_configure_client_0.csv
ceph-mgr_ceph-pprakash-jue5j3-node1-installer_setup_pre-requisites_0.csv
ceph-mon_ceph-pprakash-jue5j3-node1-installer_Deploy_cluster_using_cephadm_0.csv
ceph-mon_ceph-pprakash-jue5j3-node1-installer_Executes_RGW,_RBD_and_FS_operations_0.csv
ceph-mon_ceph-pprakash-jue5j3-node1-installer_configure_client_0.csv
ceph-mon_ceph-pprakash-jue5j3-node1-installer_setup_pre-requisites_0.csv
ceph-osd_ceph-pprakash-jue5j3-node1-installer_Deploy_cluster_using_cephadm_0.csv
ceph-osd_ceph-pprakash-jue5j3-node1-installer_Executes_RGW,_RBD_and_FS_operations_0.csv
ceph-osd_ceph-pprakash-jue5j3-node1-installer_configure_client_0.csv
ceph-osd_ceph-pprakash-jue5j3-node1-installer_setup_pre-requisites_0.csv

**Comment by pranavprakash20** [comment]
[root@ceph-pprakash-jue5j3-node1-installer ~]# cat ceph-osd_ceph-pprakash-jue5j3-node1-installer_Executes_RGW,_RBD_and_FS_operations_0.csv
"ceph-pprakash-jue5j3-node1-installer_Executes_RGW,_RBD_and_FS_operations_0",,,
Time stamp,Process Name,Process ID,CPU Usage,Memory Usage
[2024-14-02 13:59:05],ceph-osd,203726,0.6,105.84
[2024-14-02 13:59:05],ceph-osd,204881,0.6,104.715
[2024-14-02 13:59:05],ceph-osd,205845,0.6,101.531
[2024-14-02 13:59:05],ceph-osd,206966,0.5,101.691
[2024-14-02 13:59:05],ceph-osd,207948,0.7,108.539
[2024-14-02 13:59:05],ceph-osd,209080,0.5,99.0391
[2024-14-02 13:59:35],ceph-osd,203726,0.6,106.898
[2024-14-02 13:59:35],ceph-osd,204881,0.6,105.977
[2024-14-02 13:59:35],ceph-osd,205845,0.6,104.773
[2024-14-02 13:59:35],ceph-osd,206966,0.5,104.359
[2024-14-02 13:59:35],ceph-osd,207948,0.7,109.465
[2024-14-02 13:59:35],ceph-osd,209080,0.5,102.395
[2024-14-02 14:00:05],ceph-osd,203726,0.6,113.094
[2024-14-02 14:00:05],ceph-osd,204881,0.7,110.664
[2024-14-02 14:00:05],ceph-osd,205845,0.6,110.312
[2024-14-02 14:00:05],ceph-osd,206966,0.6,111.379
[2024-14-02 14:00:05],ceph-osd,207948,0.7,116.023
[2024-14-02 14:00:05],ceph-osd,209080,0.7,110.918
[2024-14-02 14:00:35],ceph-osd,203726,0.9,141.395
[2024-14-02 14:00:35],ceph-osd,204881,0.8,126.887
[2024-14-02 14:00:35],ceph-osd,205845,0.7,114.594
[2024-14-02 14:00:35],ceph-osd,206966,0.8,131.367
[2024-14-02 14:00:35],ceph-osd,207948,1.1,158.062
[2024-14-02 14:00:35],ceph-osd,209080,0.9,138.828
[2024-14-02 14:01:05],ceph-osd,203726,1.2,173.098
[2024-14-02 14:01:05],ceph-osd,204881,1.0,151.695
[2024-14-02 14:01:05],ceph-osd,205845,0.7,125.121
[2024-14-02 14:01:05],ceph-osd,206966,1.1,150.355
[2024-14-02 14:01:05],ceph-osd,207948,1.4,177.145
[2024-14-02 14:01:05],ceph-osd,209080,1.2,161.445
[2024-14-02 14:01:35],ceph-osd,203726,1.3,186.773
[2024-14-02 14:01:35],ceph-osd,204881,1.1,162.332
[2024-14-02 14:01:35],ceph-osd,205845,0.8,129.129
[2024-14-02 14:01:35],ceph-osd,206966,1.1,162.055
[2024-14-02 14:01:35],ceph-osd,207948,1.5,188.184
[2024-14-02 14:01:35],ceph-osd,209080,1.3,168.891
[2024-14-02 14:02:05],ceph-osd,203726,1.4,216.637
[2024-14-02 14:02:05],ceph-osd,204881,1.2,198.82
[2024-14-02 14:02:05],ceph-osd,205845,0.9,172.281
[2024-14-02 14:02:05],ceph-osd,206966,1.2,197.695
[2024-14-02 14:02:05],ceph-osd,207948,1.7,249.211
[2024-14-02 14:02:05],ceph-osd,209080,1.5,218.68
[2024-14-02 14:02:35],ceph-osd,203726,1.8,304.594
[2024-14-02 14:02:35],ceph-osd,204881,1.5,222.797
[2024-14-02 14:02:35],ceph-osd,205845,1.1,166.789
[2024-14-02 14:02:35],ceph-osd,206966,1.5,228.09
[2024-14-02 14:02:35],ceph-osd,207948,1.9,256.418
[2024-14-02 14:02:35],ceph-osd,209080,1.8,243.184
[2024-14-02 14:03:05],ceph-osd,203726,1.9,315.133
[2024-14-02 14:03:05],ceph-osd,204881,1.6,233.5
[2024-14-02 14:03:05],ceph-osd,205845,1.2,176.461
[2024-14-02 14:03:05],ceph-osd,206966,1.7,242.375
[2024-14-02 14:03:05],ceph-osd,207948,2.0,264.426
[2024-14-02 14:03:05],ceph-osd,209080,2.0,256.832
[2024-14-02 14:03:36],ceph-osd,203726,1.9,324.598
[2024-14-02 14:03:36],ceph-osd,204881,1.7,240.789
[2024-14-02 14:03:36],ceph-osd,205845,1.2,181.516
[2024-14-02 14:03:36],ceph-osd,206966,1.7,245.105
[2024-14-02 14:03:36],ceph-osd,207948,2.1,271.082
[2024-14-02 14:03:36],ceph-osd,209080,2.1,264.82
[2024-14-02 14:04:06],ceph-osd,203726,2.0,342.367
[2024-14-02 14:04:06],ceph-osd,204881,1.7,249.242
[2024-14-02 14:04:06],ceph-osd,205845,1.3,198.98
[2024-14-02 14:04:06],ceph-osd,206966,1.7,251.449
[2024-14-02 14:04:06],ceph-osd,207948,2.2,278.84
[2024-14-02 14:04:06],ceph-osd,209080,2.3,275.371
[2024-14-02 14:04:36],ceph-osd,203726,2.0,361.297
[2024-14-02 14:04:36],ceph-osd,204881,1.7,259.102
[2024-14-02 14:04:36],ceph-osd,205845,1.3,211.348
[2024-14-02 14:04:36],ceph-osd,206966,1.7,266.637
[2024-14-02 14:04:36],ceph-osd,207948,2.2,316.859
[2024-14-02 14:04:36],ceph-osd,209080,2.3,306.75
[2024-14-02 14:05:06],ceph-osd,203726,2.1,353.777
[2024-14-02 14:05:06],ceph-osd,204881,1.8,261.332
[2024-14-02 14:05:06],ceph-osd,205845,1.3,204.207
[2024-14-02 14:05:06],ceph-osd,206966,1.8,258.496
[2024-14-02 14:05:06],ceph-osd,207948,2.4,295.777
[2024-14-02 14:05:06],ceph-osd,209080,2.5,294.023
[2024-14-02 14:05:36],ceph-osd,203726,2.1,357.91
[2024-14-02 14:05:36],ceph-osd,204881,1.8,264.031
[2024-14-02 14:05:36],ceph-osd,205845,1.4,205.742
[2024-14-02 14:05:36],ceph-osd,206966,1.8,261.559
[2024-14-02 14:05:36],ceph-osd,207948,2.4,300.914
[2024-14-02 14:05:36],ceph-osd,209080,2.5,298.984
[2024-14-02 14:06:06],ceph-osd,203726,2.2,358.5
[2024-14-02 14:06:06],ceph-osd,204881,1.8,265.727
[2024-14-02 14:06:06],ceph-osd,205845,1.4,206.906
[2024-14-02 14:06:06],ceph-osd,206966,1.8,263.887
[2024-14-02 14:06:06],ceph-osd,207948,2.4,302.789
[2024-14-02 14:06:06],ceph-osd,209080,2.5,302.164
[2024-14-02 14:06:36],ceph-osd,203726,2.2,360.969
[2024-14-02 14:06:36],ceph-osd,204881,1.8,267.398
[2024-14-02 14:06:36],ceph-osd,205845,1.4,207.965
[2024-14-02 14:06:36],ceph-osd,206966,1.8,266.359
[2024-14-02 14:06:36],ceph-osd,207948,2.5,304.758
[2024-14-02 14:06:36],ceph-osd,209080,2.5,302.363
[2024-14-02 14:07:06],ceph-osd,203726,2.2,362.566
[2024-14-02 14:07:06],ceph-osd,204881,1.8,267.988
[2024-14-02 14:07:06],ceph-osd,205845,1.5,210.195
[2024-14-02 14:07:06],ceph-osd,206966,1.8,267.746
[2024-14-02 14:07:06],ceph-osd,207948,2.5,306.637
[2024-14-02 14:07:06],ceph-osd,209080,2.5,303.68
[2024-14-02 14:07:36],ceph-osd,203726,2.2,376.176
[2024-14-02 14:07:36],ceph-osd,204881,1.9,286.289
[2024-14-02 14:07:36],ceph-osd,205845,1.5,221.184
[2024-14-02 14:07:36],ceph-osd,206966,1.9,265.324
[2024-14-02 14:07:36],ceph-osd,207948,2.5,366.004
[2024-14-02 14:07:36],ceph-osd,209080,2.6,337.703
[2024-14-02 14:08:06],ceph-osd,203726,2.2,359.977
[2024-14-02 14:08:06],ceph-osd,204881,1.9,264.355
[2024-14-02 14:08:06],ceph-osd,205845,1.5,198.633
[2024-14-02 14:08:06],ceph-osd,206966,1.9,269.941
[2024-14-02 14:08:06],ceph-osd,207948,2.5,305.414
[2024-14-02 14:08:06],ceph-osd,209080,2.7,317.273
[2024-14-02 14:08:36],ceph-osd,203726,2.2,357.055
[2024-14-02 14:08:36],ceph-osd,204881,1.9,267.34
[2024-14-02 14:08:36],ceph-osd,205845,1.4,199.664
[2024-14-02 14:08:36],ceph-osd,206966,1.9,264.012
[2024-14-02 14:08:36],ceph-osd,207948,2.5,303.926
[2024-14-02 14:08:36],ceph-osd,209080,2.9,307.332
[2024-14-02 14:09:06],ceph-osd,203726,2.2,360.816
[2024-14-02 14:09:06],ceph-osd,204881,2.0,270.758
[2024-14-02 14:09:06],ceph-osd,205845,1.4,199.977
[2024-14-02 14:09:06],ceph-osd,206966,1.9,267.059
[2024-14-02 14:09:06],ceph-osd,207948,2.6,307.762
[2024-14-02 14:09:06],ceph-osd,209080,2.9,310.621
[2024-14-02 14:09:36],ceph-osd,203726,2.2,365.457
[2024-14-02 14:09:36],ceph-osd,204881,2.0,272.992
[2024-14-02 14:09:36],ceph-osd,205845,1.4,199.789
[2024-14-02 14:09:36],ceph-osd,206966,2.0,268.395
[2024-14-02 14:09:36],ceph-osd,207948,2.6,312.219
[2024-14-02 14:09:36],ceph-osd,209080,3.0,316.391
[2024-14-02 14:10:06],ceph-osd,203726,2.2,370.84
[2024-14-02 14:10:06],ceph-osd,204881,2.0,279.125
[2024-14-02 14:10:06],ceph-osd,205845,1.4,199.586
[2024-14-02 14:10:06],ceph-osd,206966,2.0,271.074
[2024-14-02 14:10:06],ceph-osd,207948,2.6,316.375
[2024-14-02 14:10:06],ceph-osd,209080,3.0,323.164
[2024-14-02 14:10:36],ceph-osd,203726,2.2,367.887
[2024-14-02 14:10:36],ceph-osd,204881,2.0,276.953
[2024-14-02 14:10:36],ceph-osd,205845,1.4,198.859
[2024-14-02 14:10:36],ceph-osd,206966,2.0,278.281
[2024-14-02 14:10:36],ceph-osd,207948,2.6,334.48
[2024-14-02 14:10:36],ceph-osd,209080,3.1,348.523
[2024-14-02 14:11:06],ceph-osd,203726,2.2,368.371
[2024-14-02 14:11:06],ceph-osd,204881,2.0,278.91
[2024-14-02 14:11:06],ceph-osd,205845,1.4,198.223
[2024-14-02 14:11:06],ceph-osd,206966,1.9,272.379
[2024-14-02 14:11:06],ceph-osd,207948,2.6,323.203
[2024-14-02 14:11:06],ceph-osd,209080,3.0,329.059

File: cli/performance/utilities/memory_and_cpu_logger.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        'strftime("[%Y-%d-%m %H:%M:%S]", '
+        "systime(), 1), $2,$3,$6/1024}'"
+    )
+    memory_and_cpu_consumed = run_command(cmd)
+    return memory_and_cpu_consumed
+
+
+def main():
+    """
+    Main function of the tool.
```
Comment (vamahaja) [inline cli/performance/utilities/memory_and_cpu_logger.py:48]: Add description for script parameter

**Comment by pranavprakash20** [comment]
@vamahaja Fixed the review comment.

Now the report is copied to log-dir in the format
<performance-metrics>/<testcase_name>


pranavprakash@pranavs-MacBook-Pro work % ls performance-metrics 
Sample_Test_0
pranavprakash@pranavs-MacBook-Pro work % ls performance-metrics/Sample_Test_0 
ceph-mgr-ceph-pprakash-jd878v-node1-installer-Sample_Test_0.csv	ceph-mon-ceph-pprakash-jd878v-node3-Sample_Test_0.csv
ceph-mgr-ceph-pprakash-jd878v-node2-Sample_Test_0.csv		ceph-osd-ceph-pprakash-jd878v-node1-installer-Sample_Test_0.csv
ceph-mon-ceph-pprakash-jd878v-node1-installer-Sample_Test_0.csv	ceph-osd-ceph-pprakash-jd878v-node2-Sample_Test_0.csv
ceph-mon-ceph-pprakash-jd878v-node2-Sample_Test_0.csv		ceph-osd-ceph-pprakash-jd878v-node3-Sample_Test_0.csv

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by vamahaja — APPROVED** [review]

**Pull request merged**